src/components/toast/toast.scss

```diff
@@ -15,7 +15,6 @@ $toast-width: 300px !default;
   right: $toast-anim-end;
   top: $toast-top-position;
   width: $toast-width;
-  z-index: $z-toast;
```

Comment:
The Toast needs to be higher in `z-index` than dialogs. It should probably sit above dialogs regardless of where it is rendered.